src/features/auth/auth.service.ts

```diff
@@ -11,17 +13,26 @@ export class AuthService {
 
   async validateUser(username: string, pass: string): Promise<any> {
     const user = await this._usersService.findOne(username);
-    if (user && user.password === pass) {
+    if (user && await bcrypt.compare(pass, user.password)) {
       const { password, ...result } = user;
       return result;
     }
     return null;
   }
 
   async login(user: any) {
-    const payload = { username: user.username, sub: user.userId };
-    return {
-      access_token: this._jwtService.sign(payload),
-    };
+    let obj = await this._usersService.getUserStatus(user.username);
```

Comment:
Please use variable name that describes it's job, as `obj` is very general and confusing name